src/Nethermind/Nethermind.Trie/Pruning/TinyTreePath.cs

```diff
@@ -0,0 +1,43 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using Nethermind.Core.Crypto;
+
+namespace Nethermind.Trie;
+
+/// <summary>
+/// Like TreePath, but tiny. Fit in 8 byte, like a long. Can only represent 14 nibble.
+/// </summary>
+public struct TinyTreePath
```

Comment:
The fact that `TreePath` and `TinyTreePath` store value in themselves makes this worth it to differentiate it as they are different in size. Paprika took a different approach, delegating the path content to external memory (it holds only ref to it https://github.com/NethermindEth/Paprika/blob/24a15969ef37729db040857dec312fad774d43e2/src/Paprika/Data/NibblePath.cs#L32 ). I'm wondering whether it in the future we'll include `SmallTreePath` or `NotSoBigPath` 😉 

Jokes aside, if the underlying value was delegated elsewhere (NibblePath-way), it would make these cases non-distingushable and one could have one path to rule them all. The disadvantage is the memory ownership and usage scenario, where with `ref struct` you can't copy much and pass it to another thread. Slicing, is also almost a no-op with ref struct, while here you need to copy (one could argue whether ref + length is not copying).

I'm generally not sure how this will evolve in the future.

Reply:
I know. I don't think this will be use in general outside of live pruning where memory can be a problem. I tried to make TreePath used more in #6664, but there is a slice in the middle of the traversal. Hard to know if its worth the trouble although deserialization is faster. Maybe It'll make snap server faster marginally. Did not try it out.

Reply:
My comment was not focused on performance but more on semantics and where we can go with it. In regards to the slicing, the ref-based approach in Paprika proved itself to be easy to handle with https://github.com/NethermindEth/Paprika/blob/24a15969ef37729db040857dec312fad774d43e2/src/Paprika/Data/NibblePath.cs#L197-L206 . The appends, again, they cost the copy on each append. Not sure if I can add more here. For some reason, the path being `readonly struct` somehow fits my understanding of the path. There are caveats though there.

Reply:
Ok, so I guess... its fine?

Reply:
If you, as the author, think that the current shape is good, we're good. I wanted to bring more to the table and discuss other choices taken elsewhere. I believe we're fighting mostly for the speed here so good to go from my end. 
`Resolve conversation` if you agree 🙂